datafusion/expr/src/expr_rewriter/mod.rs

```diff
@@ -303,9 +303,11 @@ pub struct NamePreserver {
     use_alias: bool,
 }
 
+type QualifiedName = (Option<TableReference>, String);
```

Comment:
I would personally suggest creating an enum to make this more explicit (rather than two level of options)-- perhaps something like

```rust
pub enum SavedName {
  ///  name is not preserved
  None, 
  /// qualified name is preserved
  Saved {
    relation: QualifiedName,
    name: String,
  }
}
```

Reply:
Done.